src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionService.cs

```diff
@@ -46,6 +65,29 @@ public async Task<List<DataSetVersionStatusSummary>> GetStatusesForReleaseVersio
             .AsNoTracking()
             .AnyAsync(dsv => dsv.ReleaseFileId == releaseFileId, cancellationToken);
     }
+
+    private async Task<Either<ActionResult, DataSetVersion>> GetDataSetVersion(Guid dataSetVersionId, CancellationToken cancellationToken)
+    {
+        return await publicDataDbContext.DataSetVersions
+           .AsNoTracking()
+           .Where(dsv => dsv.Id == dataSetVersionId)
+           .SingleOrNotFoundAsync(cancellationToken);
+    }
+
+    private Either<ActionResult, Unit> CheckCanDeleteDataSetVersion(DataSetVersion dataSetVersion)
+    {
+        if (dataSetVersion.CanBeDeleted)
+        {
+            return Unit.Instance;
+        }
+
+        return ValidationUtils.ValidationResult(new ErrorViewModel
+        {
+            Code = ValidationMessages.DataSetVersionCanNotBeDeleted.Code,
+            Message = ValidationMessages.DataSetVersionCanNotBeDeleted.Message,
+            Detail = new InvalidErrorDetail<Guid>(dataSetVersion.Id)
+        });
+    }
```

Comment:
I think we are duplicating this check in both the Admin and the Processor which isn't necessarily a problem but in https://github.com/dfe-analytical-services/explore-education-statistics/pull/4917 we have moved the Admin sides validation for creating a data set version to the Processor, so I'm thinking we should probably do the same here and make this nothing more than a proxy for the delete call. What do you think?

Reply:
I've stopped the admin from doing any other checks than checking they are a BAU user. I've adjusted the integration tests to reflect this too